packages/api/src/command/medical/patient/append-bulk-doc-download-progress.ts

```diff
@@ -0,0 +1,69 @@
+import { DocumentDownloadStatus } from "../../../domain/medical/document-bulk-download";
+import { Patient } from "../../../domain/medical/patient";
+import { PatientModel } from "../../../models/medical/patient";
+import { executeOnDBTx } from "../../../models/transaction-wrapper";
+import { getPatientOrFail } from "./get-patient";
+
+export type SetDocBulkDownloadProgress = {
+  patient: Pick<Patient, "id" | "cxId">;
+  successful?: number;
+  errors?: number;
+  status?: DocumentDownloadStatus;
+  requestId?: string;
+};
+
+/**
+ * The function `appendDocBulkDownloadProgress` updates the progress of a bulk document download to a
+ * patient's data.
+ * @param {SetDocBulkDownloadProgress}  - - A cmd argument type to update the progress of a bulk document download.
+ * @returns a Promise that resolves to a Patient object.
+ */
+export async function appendDocBulkDownloadProgress({
+  patient,
```

Comment:
Suggestion from learning after an incident this week: decompose `patient: { id, cxId }` to avoid `patient` being used on the rest of the function.

I know in this case we're already "picking" only these fields, so one can't do much with it, but leaving the suggestion for a pattern - "if we can decompose and avoid multiple constants with similar names, best to do it".